crates/re_viewport/src/container.rs

```diff
@@ -344,24 +350,40 @@ impl ContainerBlueprint {
                 egui_tiles::Container::Tabs(tabs)
             }
             egui_tiles::ContainerKind::Horizontal | egui_tiles::ContainerKind::Vertical => {
-                let linear_dir = match self.container_kind {
-                    egui_tiles::ContainerKind::Horizontal => egui_tiles::LinearDir::Horizontal,
-                    egui_tiles::ContainerKind::Vertical => egui_tiles::LinearDir::Vertical,
-                    _ => unreachable!(),
-                };
-                let mut linear = egui_tiles::Linear::new(linear_dir, children.clone());
+                match self.container_kind {
+                    egui_tiles::ContainerKind::Horizontal => {
+                        let mut linear = egui_tiles::Linear::new(
+                            egui_tiles::LinearDir::Horizontal,
+                            children.clone(),
+                        );
+
+                        for (share, id) in self.col_shares.iter().zip(children.iter()) {
+                            linear.shares[*id] = *share;
```

Comment:
~This looks very crashy to me. `linear.shares` should be empty at this point, and so `[*id]` will panic.~

**EDIT:** reading my code for `Shares` I see that `IndexMut` actually acts different from that of `HashMap` or `BTreeMap`. Still, better not to trip up the reader.

```suggestion
                            linear.shares.set_share(*id, *share);
```